packages/shorebird_cli/lib/src/os/operating_system_interface.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/platform.dart';
+import 'package:shorebird_cli/src/process.dart';
+
+// TODO(bryanoltman): remove this once os is used.
+// coverage:ignore-start
+/// A reference to a [OperatingSystemInterface] instance.
+final osRef = create(OperatingSystemInterface.new);
+
+/// The [OperatingSystemInterface] instance available in the current zone.
+OperatingSystemInterface get os => read(osRef);
+// coverage:ignore-end
+
+/// {@template operating_system_interface}
+/// A wrapper around operating system specific functionality.
+/// {@endtemplate}
+abstract class OperatingSystemInterface {
```

Comment:
Can we rename this to `OperatingSystem`?

Reply:
`flutter_tools` calls their version of this `OperatingSystemUtils`, which seems about the same level of fine to me as `OperatingSystemInterface` (although I think I might like "interface" better, because we're "interfacing" with the OS, and util is not especially descriptive in any context)

Reply:
Ah I see when I read `OperatingSystemInterface` I thought of interface as in a class that must be implemented. I think it's a little confusing to the ref `os` then we should be consistent imo 🤷

Reply:
Updated the name of `os` to `osInterface`